directives/abstract_exercise.py

```diff
@@ -51,3 +52,66 @@ def set_assistant_permissions(self, data):
 
         if 'allow-assistant-viewing' in self.options:
             data['allow_assistant_viewing'] = str_to_bool(self.options['allow-assistant-viewing'])
+
+class ConfigurableExercise(AbstractExercise):
+    option_spec = {
+        'configure-url': directives.unchanged,
+        'category': directives.unchanged,
+        'no-configure': directives.flag,
+    }
+
+    def apply_override(self, data, category = None):
+        if 'no-override' in self.options:
+            return
+
+        if not category:
+            if "category" in data:
+                category = data["category"]
+            elif "category" in self.options:
+                category = self.options["category"]
+
+        if not category:
+            return
+
+        env = self.state.document.settings.env
+        override = env.config.override
+        if category in override:
+            data.update(override[category])
+
+    def set_url(self, data, name):
+        env = self.state.document.settings.env
+
+        if "url" not in data and env.config.default_exercise_url:
+            data["url"] = env.config.default_exercise_url
+
+        data['url'] = data['url'].format(key=name)
+
+    def set_configure(self, data, exercise_url, files):
+        if 'no-configure' in self.options:
+            return
+
+        if "configure-url" in data:
```

Comment:
configure-url isn't actually supposed to be written to the YAML file: that line exists so that configure-url can be overridden using the category and the override config value. I'll replace the following line with .pop() to remove it from the output.